server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeAction.java

```diff
@@ -344,4 +372,244 @@ private static ComponentTreeWsRequest toComponentTreeWsRequest(Request request)
+  }
+
+  @CheckForNull
+  private Long searchDeveloperId(DbSession dbSession, ComponentTreeRequest wsRequest) {
+    if (wsRequest.getDeveloperId() == null && wsRequest.getDeveloperKey() == null) {
+      return null;
+    }
+
+    return componentFinder.getByUuidOrKey(dbSession, wsRequest.getDeveloperId(), wsRequest.getDeveloperKey(), DEVELOPER_ID_AND_KEY).getId();
+  }
+
+  private Map<String, ComponentDto> searchReferenceComponentsById(DbSession dbSession, List<ComponentDto> components) {
+    List<String> referenceComponentUUids = components.stream()
+            .map(ComponentDto::getCopyResourceUuid)
+            .filter(Objects::nonNull)
+            .collect(MoreCollectors.toList(components.size()));
+    if (referenceComponentUUids.isEmpty()) {
+      return emptyMap();
+    }
+
+    return FluentIterable.from(dbClient.componentDao().selectByUuids(dbSession, referenceComponentUUids))
+            .uniqueIndex(ComponentDto::uuid);
+  }
+
+  private List<ComponentDto> searchComponents(DbSession dbSession, ComponentTreeQuery componentTreeQuery) {
+    Collection<String> qualifiers = componentTreeQuery.getQualifiers();
+    if (qualifiers != null && qualifiers.isEmpty()) {
+      return Collections.emptyList();
+    }
+    return dbClient.componentDao().selectDescendants(dbSession, componentTreeQuery);
+  }
+
+  private List<MetricDto> searchMetrics(DbSession dbSession, ComponentTreeRequest request) {
+    List<String> metricKeys = requireNonNull(request.getMetricKeys());
+    List<MetricDto> metrics = dbClient.metricDao().selectByKeys(dbSession, metricKeys);
+    if (metrics.size() < metricKeys.size()) {
+      List<String> foundMetricKeys = Lists.transform(metrics, MetricDto::getKey);
+      Set<String> missingMetricKeys = Sets.difference(
+              new LinkedHashSet<>(metricKeys),
+              new LinkedHashSet<>(foundMetricKeys));
+
+      throw new NotFoundException(format("The following metric keys are not found: %s", COMMA_JOINER.join(missingMetricKeys)));
+    }
+    String forbiddenMetrics = metrics.stream()
+            .filter(metric -> ComponentTreeAction.FORBIDDEN_METRIC_TYPES.contains(metric.getValueType()))
+            .map(MetricDto::getKey)
+            .sorted()
+            .collect(MoreCollectors.join(COMMA_JOINER));
+    checkArgument(forbiddenMetrics.isEmpty(), "Metrics %s can't be requested in this web service. Please use api/measures/component", forbiddenMetrics);
+    return metrics;
+  }
+
+  private Table<String, MetricDto, ComponentTreeData.Measure> searchMeasuresByComponentUuidAndMetric(DbSession dbSession, ComponentDto baseComponent,
+                                                                                                     ComponentTreeQuery componentTreeQuery,
+                                                                                                     List<ComponentDto> components, List<MetricDto> metrics, @Nullable Long developerId) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 186 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)